arangod/Pregel/Conductor.cpp

```diff
@@ -165,56 +167,99 @@ bool Conductor::_startGlobalStep() {
   _statistics.resetActiveCount();
   _totalVerticesCount = 0;  // might change during execution
   _totalEdgesCount = 0;
-  // we are explicitly expecting an response containing the aggregated
-  // values as well as the count of active vertices
-  int res = _sendToAllDBServers(Utils::prepareGSSPath, b, [&](VPackSlice const& payload) {
+
+  VPackBuilder messagesFromWorkers;
+
+  {
+    // TODO: lock?
```

Comment:
There is no locking needed because the lambda is executed in a single thread only. Although it might be a different thread. However this access is synchronized with a `future::wait()`.